modules/pel/peltool/parse_user_data.py

```diff
@@ -40,25 +40,35 @@ def parse(self) -> str:
         else:
             value = self.parseCustom()
 
+        # Catch if value is None, otherwise python crashes
+        if value == None:
+            value = ("PROBLEM with UNRECOGNIZED value from parser creatorID={} compID={} subType={} version={}"
+                     .format(self.creatorID, hex(self.compID), self.subType, self.version))
+            return json.dumps(value)
         return value
 
     def parseCustom(self) -> str:
         import importlib
         name = (self.creatorID.lower() + "%04X" % self.compID).lower()
         try:
             cls = importlib.import_module("udparsers." + name + "." + name)
-            mv = memoryview(self.data)
-            return cls.parseUDToJson(self.subType, self.version, mv)
+            if self.data:
+                mv = memoryview(self.data)
+                return cls.parseUDToJson(self.subType, self.version, mv)
         except ImportError:
-            mv = memoryview(self.data)
-            return json.dumps(hexdump(mv))
+            print("Failed parseCustom ImportError name={} creatorID={} compID={}"
+                  .format(name, self.creatorID, hex(self.compID)))
+            if self.data:
+                mv = memoryview(self.data)
+                return json.dumps(hexdump(mv))
         except Exception as e:
             print('Failed parsing user data for creator {} compID {} '
                   'subType {} version {}: {}'.format(
                       self.creatorID, "0x%04X" % self.compID,
-                      "0x%X" % self.subType, self.version, str(e)), file=sys.stderr)
-            mv = memoryview(self.data)
-            return json.dumps(hexdump(mv))
+                      "0x%X" % self.subType, self.version, str(e)))
```

Comment:
why did you remove printing to stderr?

Reply:
By printing to stderr, the output is orphaned if the PEL output is reviewed (like reviewing PELs directed to files, etc).

Reply:
It's supposed to be.  The output is in JSON and is supposed to be machine readable.  Writing plain text will break that.

Reply:
The print output was being routed to stderr, the change allows the stdout information be dumped to stdout, which will be captured by any re-direction.

Reply:
We can't break the JSON output to stdout, or we will crash people's tools.

How about you do what I suggested in the other spot and add the error into the JSON output along with the hex dump?

Reply:
Done